awx/ui_next/src/util/qs.js

```diff
@@ -1,54 +1,163 @@
+/**
+ * helper function used to convert from
+ * Object.entries format ([ [ key, value ], ... ]) to object
+ * @param {array} array in the format [ [ key, value ], ...]
+ * @return {object} object in the forms { key: value, ... }
+ */
+const toObject = entriesArr =>
+  entriesArr.reduce((acc, [key, value]) => {
+    if (acc[key] && Array.isArray(acc[key])) {
+      acc[key].push(value);
+    } else if (acc[key]) {
+      acc[key] = [acc[key], value];
+    } else {
+      acc[key] = value;
+    }
+    return acc;
+  }, {});
+
+/**
+ * helper function to namespace params object
+ * @param {string} namespace to append to params
+ * @param {object} params object to append namespace to
+ * @return {object} params object with namespaced keys
+ */
+const namespaceParams = (namespace, params = {}) => {
+  if (!namespace) return params;
+
+  const namespaced = {};
+  Object.keys(params).forEach(key => {
+    namespaced[`${namespace}.${key}`] = params[key];
+  });
+
+  return namespaced || {};
+};
+
+/**
+ * helper function to remove namespace from params object
+ * @param {string} namespace to remove from params
+ * @param {object} params object to append namespace to
+ * @return {object} params object with non-namespaced keys
+ */
+const denamespaceParams = (namespace, params = {}) => {
+  if (!namespace) return params;
+
+  const denamespaced = {};
+  Object.keys(params).forEach(key => {
+    denamespaced[key.substr(namespace.length + 1)] = params[key];
+  });
+
+  return denamespaced || {};
+};
+
+/**
+ * helper function to check the namespace of a param is what you expec
+ * @param {string} namespace to append to params
+ * @param {object} params object to append namespace to
+ * @return {object} params object with namespaced keys
+ */
+const namespaceMatches = (namespace, fieldname) => {
+  if (!namespace) return !fieldname.includes('.');
+
+  return fieldname.startsWith(`${namespace}.`);
+};
+
+/**
+ * helper function to check the value of a param is equal to another
+ * @param {string or number or array} param value one
+ * @param {string or number or array} params value two
+ * @return {boolean} true if values are equal
+ */
+const paramValueIsEqual = (one, two) => {
+  let isEqual = false;
+
+  if (Array.isArray(one) && Array.isArray(two)) {
+    isEqual = one.filter(val => two.indexOf(val) > -1).length === 0;
+  } else if (
+    (typeof one === 'string' && typeof two === 'string') ||
+    (typeof one === 'number' && typeof two === 'number')
+  ) {
+    isEqual = one === two;
+  }
+
+  return isEqual;
+};
+
```

Comment:
You can probably get rid of `isEqual` var here and just return early in each of the if blocks